ref: add doc for pessimistic auto-commit (#7868)
pingcap/docs: tidb-configuration-file.md

### deadlock-history-collect-retryable

+ Controls whether the [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) table collects the information of retryable deadlock errors. For the description of retryable deadlock errors, see [Retryable deadlock errors](/information-schema/information-schema-deadlocks.md#retryable-deadlock-errors).

> ### pessimistic-auto-commit (New in v6.0)

Reviewer: ```suggestion
### pessimistic-auto-commit (New in v6.0.0)
```

> + Determines the transaction mode that the auto-commit transaction uses when the pessimistic transaction mode is globally enabled (`tidb_txn_mode='pessimistic'`). By default, even if the pessimistic transaction mode is globally enabled, the auto-commit transaction still uses the optimistic transaction mode. When `pessimistic-auto-commit` is enabled (set to `true`), the auto-commit transaction also uses pessimistic mode, which is consistent with the other explicitly committed pessimistic transactions.

Reviewer: ```suggestion
+ Determines the transaction mode that an auto-commit transaction uses when the pessimistic transaction mode is globally enabled (`tidb_txn_mode='pessimistic'`). By default, even if the pessimistic transaction mode is globally enabled, auto-commit transactions still use the optimistic transaction mode. When `pessimistic-auto-commit` is enabled (set to `true`), the auto-commit transactions also use pessimistic mode as other explicitly committed pessimistic transactions.
```

> + For scenarios with conflicts, after enabling this configuration, TiDB includes the auto-commit transactions into the global lock-waiting management, which avoids dead locks and mitigates the high latency brought by deadlock-causing conflicts.

Author: ```suggestion
+ For scenarios with conflicts, after enabling this configuration, TiDB includes the auto-commit transactions into the global lock-waiting management, which avoids dead locks and mitigates the latency spike brought by deadlock-causing conflicts.
```

> + For scenarios with conflicts, after enabling this configuration, TiDB includes the auto-commit transactions into the global lock-waiting management, which avoids dead locks and mitigates the latency spike brought by deadlock-causing conflicts.
> + For scenarios with no conflicts, if there are many auto-commit transactions and a single transaction operates a large data volume, enabling this configuration causes performance regression. For example, the auto-commit `INSERT INTO SELECT` statement.

Reviewer: ```suggestion
+ For scenarios with conflicts, after enabling this configuration, TiDB includes auto-commit transactions into the global lock-waiting management, which avoids deadlocks and mitigates the latency spike brought by deadlock-causing conflicts.
+ For scenarios with no conflicts, if there are many auto-commit transactions, and a single transaction operates a large data volume, enabling this configuration causes performance regression. For example, the auto-commit `INSERT INTO SELECT` statement.
```
